File: c++/HawkQtV01/Hawk/Hawk/Controller.cpp
#include "Controller.h"

void Controller::addSubject(std::string name)
{
	Subject newSubject;
	newSubject.setSize(0);
	newSubject.setName(name);
	this->repo.addSubject(newSubject);
}

void Controller::removeSubject(std::string name)
{
	Subject newSubject = this->repo.findByName(name);
	this->repo.removeSubject(newSubject);
}

void Controller::addHours(std::string name, std::string desc,int nrHours)
{
	this->repo.addHours(name, desc, nrHours);
}

void Controller::changeName(std::string oldName, std::string newName)
{
	this->repo.changeName(oldName, newName);
}

std::vector<Subject>& Controller::getVector()
{
	return this->repo.getVector();
}

Subject& Controller::findByName(std::string name)
{
	return this->repo.findByName(name);
}

void Controller::writeToFile(std::string file) {
	this->repo.writeTofile(file);
}

Controller::Controller()
{
	Repository newRepo;
	this->repo = newRepo;
}


Controller::~Controller()
{
}

File: python/Work schedule 2016/controller/Controller.py
'''
Created on 4 Feb 2016

@author: Mihai
'''

class Controller(object):
    '''
    Controlls operations regarding the repository
    '''


    def __init__(self, repository):
        '''
        Constructor
        '''
        self._repo = repository
        
    def addSubject(self,subject):
        '''
        adds a subject to the repository
        '''
        return self._repo.addSubject(subject)
    
    def removeSubject(self,name):
        '''
        removes a subject from the repository by name
        '''
        return self._repo.removeSubject(name)
    
    def filterLowest(self):
        '''
        filters the repository, returning the lowest subject (by hours invested)
        '''
        return self._repo.filterLowest()
    
    def updateSubject(self,oldSubjectName, newSubject):
        '''
        updates a subject found by name, updating both hours and/or name
        '''
        return self._repo.updateSubject(oldSubjectName, newSubject)
    
    def addHours(self,name,hours):
        '''
        adds hours to a subject by its name
        '''
        self._repo.addHours(name,hours)
    
    def getAll(self):
        return self._repo.getAll()
        